### DISCUSSION TITLE
Not able to use my production endpoint

### DISCUSSION THREAD

**Discussion by hexdecimal16** [urql-graphql/urql#2212] [Q&A] [answered]
urql version & exchanges: 2.0.6

**Steps to reproduce**

1. initialise a react native project using expo
2. use https://api.stacket.in:4000/graphql as the endpoint

**Expected behavior**
should be able to retrive data

**Actual behavior**
[CombinedError: [Network] Network request failed]

**Comment by JoviDeCroock** [comment, marked as answer]
Just tried in both

- [web](https://codesandbox.io/s/festive-elbakyan-pdmvk)
- [React-Native](https://snack.expo.dev/W-9_3rJjm)

Seems to work correctly, we'll need more information to help you out 😅

**Comment by hexdecimal16** [comment]
> Just tried in both
> 
> * [web](https://codesandbox.io/s/festive-elbakyan-pdmvk)
> * [React-Native](https://snack.expo.dev/W-9_3rJjm)
> 
> Seems to work correctly, we'll need more information to help you out sweat_smile


I am attaching my dependencies 
```

  "dependencies": {
    "@apollo/client": "^3.5.7",
    "@apollo/react-hooks": "^4.0.0",
    "@expo/metro-config": "^0.3.9",
    "@kanzitelli/if-component": "^0.3.0",
    "@react-native-async-storage/async-storage": "^1.15.14",
    "@react-native-community/clipboard": "^1.5.1",
    "@react-native-community/masked-view": "^0.1.11",
    "@react-native-community/netinfo": "7.1.3",
    "@react-native-picker/picker": "^2.2.1",
    "@react-navigation/bottom-tabs": "^6.0.9",
    "@react-navigation/native": "^6.0.6",
    "@react-navigation/native-stack": "^6.2.5",
    "@react-navigation/stack": "^6.0.11",
    "apollo-cache-inmemory": "^1.6.6",
    "apollo-client": "^2.6.10",
    "apollo-link-http": "^1.5.17",
    "aws-amplify": "^4.3.12",
    "aws-amplify-react-native": "^6.0.2",
    "date-fns": "^2.28.0",
    "expo": "^44.0.0",
    "expo-blur": "~11.0.0",
    "expo-dev-client": "~0.8.0",
    "expo-font": "~10.0.4",
    "expo-localization": "~12.0.0",
    "expo-splash-screen": "~0.14.1",
    "expo-status-bar": "~1.2.0",
    "expo-updates": "~0.11.5",
    "formik": "^2.2.9",
    "graphql": "^16.2.0",
    "graphql-tag": "^2.12.6",
    "i18n-js": "^3.8.0",
    "lodash": "^4.17.21",
    "mobx": "^6.3.12",
    "mobx-persist-store": "^1.0.4",
    "mobx-react": "^7.2.1",
    "react": "17.0.1",
    "react-apollo": "^3.1.5",
    "react-dom": "17.0.1",
    "react-native": "0.64.3",
    "react-native-confirmation-code-field": "^7.2.0",
    "react-native-gesture-handler": "~2.1.0",
    "react-native-mmkv": "1.3.2",
    "react-native-reanimated": "~2.3.1",
    "react-native-responsive-screen": "^1.4.2",
    "react-native-safe-area-context": "3.3.2",
    "react-native-screens": "~3.10.1",
    "react-native-shadow-2": "^6.0.0",
    "react-native-ui-lib": "^6.7.0",
    "react-native-web": "0.17.1",
    "rn-bounceable": "^1.1.0",
    "unique-names-generator": "^4.6.0",
    "urql": "^2.0.6",
    "yup": "^0.32.11"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9",
    "@expo/config-plugins": "^4.0.15",
    "@expo/webpack-config": "~0.16.2",
    "@react-native-community/eslint-config": "^3.0.1",
    "@types/i18n-js": "^3.8.2",
    "@types/lodash": "^4.14.178",
    "@types/react": "~17.0.21",
    "@types/react-dom": "~17.0.9",
    "@types/react-native": "~0.64.12",
    "@typescript-eslint/parser": "^5.9.0",
    "dotenv-cli": "^4.1.1",
    "prettier": "^2.4.1",
    "release-it": "^14.11.8",
    "typescript": "~4.3.5"
  }
```

**Comment by hexdecimal16** [comment]
> Just tried in both
> 
> * [web](https://codesandbox.io/s/festive-elbakyan-pdmvk)
> * [React-Native](https://snack.expo.dev/W-9_3rJjm)
> 
> Seems to work correctly, we'll need more information to help you out sweat_smile


https://snack.expo.dev/@dhairy/sandbox-react-native
Here is an example of the error. It is important to note the error only showes when making any query not while defining client

**Comment by JoviDeCroock** [comment]
The sandbox you linked just works 😅 
![image](https://user-images.githubusercontent.com/17125876/150592384-af0756b1-5966-4146-946a-3b1e9bcf1045.png)

**Comment by kitten** [comment]
Converted to a discussion since it's unlikely to be related to urql itself

**Reply by hexdecimal16** [reply]
Ok. Please try to check on actual android device issue is still there, though it works on web.

**Comment by hexdecimal16** [comment]
So I have done multiple test, library is working perfectly on web versions but giving network error on android device and emulator.

**Comment by hexdecimal16** [comment]
I am reopening this as a new issue

**Reply by kitten** [reply]
to be honest, I would have to close this as duplicate if you do 😅 With all the love and respect, we love to help, and once we get enough information about this and/or we can reproduce this we will. But a dupllicate issue won't change this, since we haven't seen similar reports or conversations and we'd still be having the above discussion in a duplicate space ❤️

**Reply by JoviDeCroock** [reply]
To echo the above, it works on iOS and web, maybe android does something odd to the less conventional url-format. Might want to try with an alternative url 😅 I would generally conclude that if it's an android specific problem that chances are very small that this is related to a JS-library.

**Reply by kitten** [reply]
We do host example APIs at `trygql.formidable.dev` so you can quickly send some test queries, e.g. https://trygql.formidable.dev/graphql/basic-pokedex
Sample React Native code here that could be dropped into a Snack: https://github.com/FormidableLabs/urql/tree/main/examples/with-react-native

**Reply by hexdecimal16** [reply]
Ok thanks same thing happens with apollo-client as well don't know what excatly is the issue. Have you ever checked on an api where introspection is turned off on the server?
@kitten @JoviDeCroock

**Reply by JoviDeCroock** [reply]
introspection being turned off isn't an issue, I would try a server that isn't specifying a port maybe android has an issue with that hence the suggestion of using one of the above API's to experiment with.

**Reply by hexdecimal16** [reply]
> introspection being turned off isn't an issue, I would try a server that isn't specifying a port maybe android has an issue with that hence the suggestion of using one of the above API's to experiment with.

I tried using ip address of my ec2 instance where graphql server is running that also didn't help. (eg. https://13.123.123.12:4000/graphql )

**Reply by JoviDeCroock** [reply]
That still uses a port though 😅

**Reply by hexdecimal16** [reply]
> That still uses a port though sweat_smile

but that works when I am running same server on my local machine and using my local machine ip address (eg. http://192.168.29.97:4000/graphql)

**Reply by kitten** [reply]
Basicallly, we believe it may be a kind of TLS/HTTPS-related error

**Reply by hexdecimal16** [reply]
It is not a port issue because https://api.stacket.in/graphql is also not working. I think it could be SSL issue.@JoviDeCroock

**Reply by hexdecimal16** [reply]
> Basicallly, we believe it may be a kind of TLS/HTTPS-related error

Any ideas ?

**Reply by JoviDeCroock** [reply]
Because you didn't want to take our suggestions which are just offered because we wanted to help in our free time, I did it for you [a reproduction that works on android with a good TLS/HTTPS cert](https://snack.expo.dev/EJ9FCrAHt) this works correctly, so the issue is located on your endpoint/certification

**Reply by hexdecimal16** [reply]
> Because you didn't want to take our suggestions which are just offered because we wanted to help in our free time, I did it for you [a reproduction that works on android with a good TLS/HTTPS cert](https://snack.expo.dev/EJ9FCrAHt) this works correctly, so the issue is located on your endpoint/certification

Thanks, I appreciated you time and efforts! Finally I was able to fix with your suggestions the issue was indeed with server setup. I was using http server with ssl keys to make it secure after removing that and using nginx the problem was solved. Thanks again!
